runtime/integration-tests/src/generic/cases/liquidity_pools.rs

```diff
@@ -3889,48 +3865,65 @@ mod development {
 				setup_test(&mut env);
 
 				env.parachain_state_mut(|| {
-					let pool_id = DEFAULT_POOL_ID;
+					let pool_id = POOL_ID;
 					let investor: AccountId = AccountConverter::<T, LocationToAccountId>::convert(
 						(DOMAIN_MOONBEAM, Keyring::Bob.into()),
 					);
 					let pool_currency: CurrencyId = AUSD_CURRENCY_ID;
 					let foreign_currency: CurrencyId = USDT_CURRENCY_ID;
 					let pool_currency_decimals = currency_decimals::AUSD;
 					let invest_amount_pool_denominated: u128 = 6 * decimals(18);
+					let trader: AccountId = Keyring::Alice.into();
 					create_currency_pool::<T>(
 						pool_id,
 						pool_currency,
 						pool_currency_decimals.into(),
 					);
 
-					// Short cut: Invest through investments to bump pending investment amount
-					do_initial_invest_short_cut::<T>(
-						pool_id,
-						invest_amount_pool_denominated,
-						investor.clone(),
-					);
-
 					// USDT investment preparations
 					let invest_amount_foreign_denominated = enable_usdt_trading::<T>(
 						pool_currency,
 						invest_amount_pool_denominated,
-						false,
 						true,
 						true,
-						|| {},
+						true,
 					);
 
-					// Should be able to invest with foreign currency as ForeignInvestmentState does
-					// not exist yet
+					// Do first investment and fulfill swap order
 					do_initial_increase_investment::<T>(
 						pool_id,
 						invest_amount_foreign_denominated,
 						investor.clone(),
 						foreign_currency,
 					);
+					let swap_order_id = pallet_foreign_investments::Swaps::<T>::swap_id_from(
+						&investor,
+						default_investment_id::<T>(),
+						pallet_foreign_investments::Action::Investment,
+					)
```

Comment:
Not sure if it could make sense to expose this under a `RuntimeAPI`. How do Apps know which order placed by FI needs to be fulfilled?

@cc @mustermeiszer

Reply:
I think it's fine to query `ForeignIdToSwapId::<T>::get((account, investment_id, action))` for Apps which would be `api.query.foreignInvestments.foreignIdToSwapId([account, investmentId, action['Investment'])`.

Reply:
Make sense!